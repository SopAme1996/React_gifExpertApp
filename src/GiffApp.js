import React, { useState } from 'react';
import { AddCategoria } from './Componentes/AddCategoria';
import { RestApi } from './Componentes/RestApi';


export const GiffApp = () => {

    const [data, setData]=useState(['One Punch']);

    return (
        <>
            <header>
                <h1>GifApp</h1>
            </header>
            <hr/>

            <section>
                <AddCategoria setData = { setData }/>
            </section>

            <section>
                {
                    data.map(nombre =>{
                        return(
                       <RestApi 
                       key={nombre}
                       categoria={nombre}
                       />
                       )
                    })
                }
            </section>
        </>
    )
}
