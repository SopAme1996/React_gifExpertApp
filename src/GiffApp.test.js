import React from 'react';
import { GiffApp } from './GiffApp';
import { shallow } from 'enzyme';

describe('Test de giffApp', () => {
    

    test('Test del componente padre de los otros componentes', () => {
        const wrapper = shallow(<GiffApp/>);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('Debe de mostrar una lista de categorias ', () => {
        const categorias = ['vegeta', 'goku'];
        const wrapper = shallow(<GiffApp defaultCategoria={categorias}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('RestApi').length).toBe(categorias.length);
    })
    
});