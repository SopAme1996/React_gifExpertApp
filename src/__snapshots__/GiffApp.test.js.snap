// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de giffApp Debe de mostrar una lista de categorias  1`] = `
<Fragment>
  <header>
    <h1>
      GifApp
    </h1>
  </header>
  <hr />
  <section>
    <AddCategoria
      setData={[Function]}
    />
  </section>
  <section>
    <RestApi
      categoria="vegeta"
      key="vegeta"
    />
    <RestApi
      categoria="goku"
      key="goku"
    />
  </section>
</Fragment>
`;

exports[`Test de giffApp Test del componente padre de los otros componentes 1`] = `
<Fragment>
  <header>
    <h1>
      GifApp
    </h1>
  </header>
  <hr />
  <section>
    <AddCategoria
      setData={[Function]}
    />
  </section>
  <section />
</Fragment>
`;
